### DISCUSSION TITLE
input and output

### DISCUSSION THREAD

**Discussion by JJun-Guo** [snakers4/silero-vad#230] [Q&A] [answered]
Hi,I input a speech sample with 44.1kHz sample rate and 16 (Sampling depth)，the output is  a speech with 16kHZ sample rate and 32  (Sampling depth), how can I get the default sampling depth speech sample???

**Comment by snakers4** [comment, marked as answer]
Hi,

The VAD accepts only 16 kHz or 8 kHz. 

https://github.com/snakers4/silero-vad/blob/7c671a75c2f82830a550ab4a182e86d3a2609d70/utils_vad.py#L119-L171

**Reply by JJun-Guo** [reply]
yes, the speech with other sample rate will be resampled with 16kHz? and the sampling depth can be choosed

**Reply by snakers4** [reply]
You need to resample.

**Reply by JJun-Guo** [reply]
I get the following code including the resample process, do i need to resample the speech manually？
def read_audio(path: str,
               sampling_rate: int = 16000):
    wav, sr = torchaudio.load(path)

    if wav.size(0) > 1:
        wav = wav.mean(dim=0, keepdim=True)

    if sr != sampling_rate:
        transform = torchaudio.transforms.Resample(orig_freq=sr,
                                                   new_freq=sampling_rate)
        wav = transform(wav)
        sr = sampling_rate
    assert sr == sampling_rate
    return wav.squeeze(0)

**Reply by snakers4** [reply]
> do i need to resample the speech manually

You need to make sure that incoming samples are either 16 kHz or 8 kHz
